### PR TITLE
Fix Never api key expiration dates

### LINKED ISSUE #4714: Api key doesn't display the good expiration date
# Bug

The expiration date display `29 days` instead of `Never`

https://github.com/twentyhq/twenty/assets/19412894/ee8222b9-b2e1-4081-b8b3-3a9fa21e2cfa

### PR DESCRIPTION
**Pull request by martmull**
closes #4714 

We cannot set null expiration dates for api keys. So we will set to 100 years instead of null. If apiKey expires in more that 10 years, it is displayed as "Never expires"

### CODE REVIEW

File: packages/twenty-front/src/modules/settings/developers/utils/format-expiration.ts [hunk 1]
```diff
@@ -11,6 +11,9 @@ export const formatExpiration = (
 ) => {
   if (isNonEmptyString(expiresAt)) {
     const dateDiff = beautifyDateDiff(expiresAt, undefined, short);
+    if (dateDiff === '+∞') {
+      return withExpiresMention ? 'Never expires' : 'Never';
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/developers/utils/format-expiration.ts:15]: I don't think we should do the condition on the beautified date, it does not seems robust? Can we do it on the expiresAt directly?

File: packages/twenty-front/src/utils/date-utils.ts [hunk 2]
```diff
@@ -117,12 +119,15 @@ export const beautifyDateDiff = (
...
   dateToCompareWith?: string,
   short = false,
+  infiniteDiffTriggerInYears = NEVER_EXPIRE_DELTA_IN_YEARS / 10,
 ) => {
   const dateDiff = DateTime.fromISO(date).diff(
     dateToCompareWith ? DateTime.fromISO(dateToCompareWith) : DateTime.now(),
     ['years', 'days'],
   );
   let result = '';
+  if (dateDiff.years && dateDiff.years >= infiniteDiffTriggerInYears)
```
Comment (charlesBochet) [inline packages/twenty-front/src/utils/date-utils.ts:129]: we should add { } around if statement content per project convention

File: packages/twenty-front/src/modules/settings/developers/utils/format-expiration.ts (hunk 1 above)
Comment (martmull) [reply packages/twenty-front/src/modules/settings/developers/utils/format-expiration.ts:15]: Yes I can create a dedicated util

File: packages/twenty-front/src/utils/date-utils.ts [hunk 3]
```diff
@@ -117,12 +119,15 @@ export const beautifyDateDiff = (
   date: string,
   dateToCompareWith?: string,
   short = false,
+  infiniteDiffTriggerInYears = NEVER_EXPIRE_DELTA_IN_YEARS / 10,
```
Comment (charlesBochet) [inline packages/twenty-front/src/utils/date-utils.ts:122]: Are we displaying +infinite somewhere? If it's only to use it in the condition mentionned above, it seems to be too hacky

File: packages/twenty-front/src/modules/settings/developers/utils/format-expiration.ts [hunk 4]
```diff
@@ -1,22 +1,32 @@
 import { isNonEmptyString } from '@sniptt/guards';
+import { DateTime } from 'luxon';
 
+import { NEVER_EXPIRE_DELTA_IN_YEARS } from '@/settings/developers/constants/NeverExpireDeltaInYears.ts';
 import { ApiFieldItem } from '@/settings/developers/types/api-key/ApiFieldItem';
 import { ApiKey } from '@/settings/developers/types/api-key/ApiKey';
 import { beautifyDateDiff } from '~/utils/date-utils';
 
+export const doesNeverExpire = (expiresAt: string) => {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/developers/utils/format-expiration.ts:9]: usually we prefer 1 util per file to enforce clear file naming

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**